modules/gateway/peers.go

```diff
@@ -207,18 +292,41 @@ func (g *Gateway) threadedAcceptConn(conn net.Conn) {
 		delete(g.peers, kick)
 		g.log.Printf("INFO: disconnected from %v to make room for %v", kick, addr)
 	}
-	// add the peer
-	g.addPeer(&peer{
-		Peer: modules.Peer{
-			NetAddress: addr,
-			Inbound:    true,
-			Version:    remoteVersion,
-		},
-		sess: muxado.Server(conn),
-	})
-	g.mu.Unlock()
+}
+
+// acceptConnPortHandshake performs the port handshake and should be called on
+// the side accepting a connection request. The remote address is only returned
+// if err == nil.
+func acceptConnPortHandshake(conn net.Conn) (remoteAddr modules.NetAddress, err error) {
+	addr := modules.NetAddress(conn.RemoteAddr().String())
+
+	// Read the peer's port that we can dial them back on.
+	var dialbackPort string
+	err = encoding.ReadObject(conn, &dialbackPort, 13) // Max port # is 65535 (5 digits long) + 8 byte string length prefix
+	if err != nil {
+		return "", fmt.Errorf("could not read remote peer's (%v) port: %v", addr, err)
```

Comment:
including the peer's address in the error here is redundant as `threadedAcceptConn` logs the peer's address along with the error.